packages/block-editor/src/components/global-styles/background-panel.js

```diff
@@ -0,0 +1,580 @@
+		) {
+			onUploadError(
+				__( 'Only images can be used as a background image.' )
+			);
+			return;
+		}
+
+		onChange(
+			setImmutably( style, [ 'background', 'backgroundImage' ], {
+				url: media.url,
+				id: media.id,
+				source: 'file',
+				title: media.title || undefined,
+			} )
+		);
+	};
+
+	const onFilesDrop = ( filesList ) => {
+		mediaUpload( {
+			allowedTypes: [ 'image' ],
+			filesList,
+			onFileChange( [ image ] ) {
+				if ( isBlobURL( image?.url ) ) {
+					return;
+				}
+				onSelectMedia( image );
+			},
+			onError: onUploadError,
+		} );
+	};
+
+	const resetAllFilter = useCallback( ( previousValue ) => {
+		return {
+			...previousValue,
+			style: {
+				...previousValue.style,
+				background: undefined,
+			},
+		};
+	}, [] );
+
+	const hasValue =
+		hasBackgroundImageValue( style ) ||
+		hasBackgroundImageValue( inheritedValue );
+
+	return (
+		<ToolsPanelItem
+			className="single-column"
+			hasValue={ () => hasValue }
+			label={ __( 'Background image' ) }
+			onDeselect={ resetBackgroundImage }
+			isShownByDefault={ isShownByDefault }
+			resetAllFilter={ resetAllFilter }
+			panelId={ panelId }
+		>
+			<div
+				className="block-editor-global-styles-background-panel__inspector-media-replace-container"
+				ref={ replaceContainerRef }
+			>
+				<MediaReplaceFlow
+					mediaId={ id }
+					mediaURL={ url }
+					allowedTypes={ [ IMAGE_BACKGROUND_TYPE ] }
+					accept="image/*"
+					onSelect={ onSelectMedia }
+					name={
+						<InspectorImagePreview
+							label={ __( 'Background image' ) }
+							filename={ title || __( 'Untitled' ) }
+							url={ url }
+						/>
+					}
+					variant="secondary"
+				>
+					{ hasValue && (
+						<MenuItem
+							onClick={ () => {
+								const [ toggleButton ] = focus.tabbable.find(
+									replaceContainerRef.current
+								);
+								// Focus the toggle button and close the dropdown menu.
+								// This ensures similar behaviour as to selecting an image, where the dropdown is
+								// closed and focus is redirected to the dropdown toggle button.
+								toggleButton?.focus();
+								toggleButton?.click();
+								resetBackgroundImage();
+							} }
+						>
+							{ __( 'Reset ' ) }
+						</MenuItem>
+					) }
+				</MediaReplaceFlow>
+				<DropZone
+					onFilesDrop={ onFilesDrop }
+					label={ __( 'Drop to upload' ) }
+				/>
+			</div>
+		</ToolsPanelItem>
+	);
+}
+
+function BackgroundSizeToolsPanelItem( {
+	panelId,
+	isShownByDefault,
+	onChange,
+	style,
+	inheritedValue,
+} ) {
+	const sizeValue =
+		style?.background?.backgroundSize ||
+		inheritedValue?.background?.backgroundSize;
+	const repeatValue =
+		style?.background?.backgroundRepeat ||
+		inheritedValue?.background?.backgroundRepeat;
+	const imageValue =
+		style?.background?.backgroundImage?.url ||
+		inheritedValue?.background?.backgroundImage?.url;
+	const positionValue =
+		style?.background?.backgroundPosition ||
+		inheritedValue?.background?.backgroundPosition;
```

Comment:
Should we care about incoming, named values from theme json, e.g., `"center center"`, or "top left"` etc.

I remember @andrewserong might have been doing some conversion in another PR, maybe unrelated to background. 🙃

Reply:
Ah, yes, we were originally looking at it when adding in `backgroundPosition` support but I dropped it as it made things a bit more complex and we were getting closer to the Beta 1 feature freeze. It'd be a good thing to add back in, but it could be its own PR.